source/type/type_utils.cpp

```diff
@@ -82,4 +82,33 @@ pbnavitia::RTLevel to_pb_realtime_level(const navitia::type::RTLevel realtime_le
             throw navitia::exception("realtime level case not handled");
     }
 }
+
+namespace type {
+
+type::DirectionType get_direction_type(pbnavitia::DirectionType direction_type) {
```

Comment:
Why not returning a `boost::optional<type::DirectionType>` instead ? That would simplify the code around this.
Same applies for the function below.

Reply:
Because i directly want to manipulate the enum :taco:

Reply:
And, I tested that the trow is well catched